src/pl/byt/Aplikacja.java

```diff
@@ -0,0 +1,55 @@
+package pl.byt;
+
+import java.util.Date;
+
+public class Aplikacja {
+    private Date dataRozpoczecia;
+    private Date dataZakonczenia;
+    private String status;
+
+    private Kandydat kandydat;
+    private RekrutacjaMediator rekrutacjaMediator;
+
+    public Aplikacja(Kandydat kandydat, RekrutacjaMediator rekrutacjaMediator) {
+        setKandydat(kandydat);
+        setRekrutacjaMediator(rekrutacjaMediator);
+    }
+
+    public void setKandydat(Kandydat kandydat) {
+        if (this.kandydat != null && this.kandydat != kandydat) {
+            kandydat.removeAplikacja(this);
+        }
```

Comment:
Don't delete, but make this no-op when the value is already set. The attribute class is tightly related to the association. If the references changes then the association no longer exists and the new one is created.